### DISCUSSION TITLE
latest code of bluesky-social/ozone nor auto-build docker image cannot work.

### DISCUSSION THREAD

**Discussion by itaru2622** [bluesky-social/atproto#2626] [Dev Debugging] [answered]
latest ozone code uses tools.ozone.server.getConfig XRPC method in
https://github.com/bluesky-social/ozone/blob/68c459aab1872765559c3b6ead42a5f938e84fd9/lib/client.ts#L172-L175

but those recent features are  not available as below.
It got error during sign-in flow to ozone, even latest ozone image ghcr.io/bluesky-social/ozone:sha-68c459aab1872765559c3b6ead42a5f938e84fd9

```
Error: Method Not Implemented
    at ServiceClient.call (2460-82b8d242f915cbd5.js:1:301076)
    at async ClientManager._getServerConfig (6281-5595efd7e229122d.js:1:6972)
    at async ClientManager.signin (6281-5595efd7e229122d.js:1:5311)
    at async onSubmit (layout-51db4003679b916c.js:1:17461)
```

```
# calling tool.ozone.team.listMembers also raises error
curl -L -X POST "https://ozone/xrpc/tools.ozone.team.listMembers" -u "admin:yourvalidpassword"
{"error":"MethodNotImplemented","message":"Method Not Implemented"}

# check package versions in latest docker images
docker run -it --rm ghcr.io/bluesky-social/ozone:sha-68c459aab1872765559c3b6ead42a5f938e84fd9 npm list -a | grep @atproto/
+-- @atproto/ozone@0.1.11
| +-- @atproto/api@0.12.7

# commit hash recorded in latest ozone docker image
docker inspect ghcr.io/bluesky-social/ozone:sha-68c459aab1872765559c3b6ead42a5f938e84fd9 | grep org.opencontainers.image.revision
        "org.opencontainers.image.revision": "68c459aab1872765559c3b6ead42a5f938e84fd9",

# double check if recent APIs are included in latest ozone docker image
docker run -it --rm ghcr.io/bluesky-social/ozone:sha-68c459aab1872765559c3b6ead42a5f938e84fd9 ls -1 node_modules/@atproto/api/dist/client/types/tools/ozone/
communication
moderation
# !!! no server nor team folders exist.
# comparing with https://github.com/bluesky-social/atproto/tree/%40atproto/api%400.12.22/packages/api/src/client/types/tools/ozone
```
while ozone/packages trying to use API version 0.12.22  in 
https://github.com/bluesky-social/ozone/blob/68c459aab1872765559c3b6ead42a5f938e84fd9/package.json#L19-L20

**Comment by itaru2622** [comment]
@foysalit It would be appreciated if you could investigate the error and fix it.

**Reply by itaru2622** [reply]
@foysalit I found workaround to use basic features with latest ozone.<br>

- update package.json and service/package.json to use latest atproto package versions by below command.
```bash
# in toplevel folder of bluesky-social/ozone
sed -i -E 's#"@atproto/api": "([0-9\.]+)"#"@atproto/api": "^\1"#'     package.json
sed -i -E 's#"@atproto/ozone": "([0-9\.]+)"#"@atproto/ozone": "^\1"#' service/package.json
```
- rebuild ozone docker image and check if latest versions are involved
```bash
docker run -it --rm ozone:latest npm list -a | grep atproto/
+-- @atproto/ozone@0.1.28
| +-- @atproto/api@0.12.23
```

**the error is still raised in sign-in process when labeler's serviceEndpoint is NOT in DID Document.**
so  add it with Bluesky API before sign-in to ozone.
finally, I could sign-in and use basic features of ozone.

former ozone supports sign-in with ordinal account (who has no labeler's serviceEndpoint in DID Document), so there is something wrong with the latest code.

**Comment by bnewbold** [comment, marked as answer]
We generally release Ozone and PDS features in a coordinated way by pushing out tagged releases. The `main` git branch is for development and won't always be functional at every point in time, as seems to be the case right now. We are hoping to get out a tagged release in the next week, but it might slip until later in July.

**Reply by itaru2622** [reply]
@bnewbold thank you for your comment.

The problem of bluesky-social/ozone repository is few TAGs to help distinguishing releasing from developing. regarding other repositories (atproto and social-app), they got tags more frequently.

additionally, the binaries of bluesky-social/ozone are still using old 0.1.1x series of atproto/ozone, there is no chance to use the recent released features in atproto/ozone@0.1.2x unless self-building binary from source.

```
# commit hash which update package.json
docker run -it --rm ghcr.io/bluesky-social/ozone:sha-7b9d5d776a7cedbe0033760ad8a2c43a69b26227 npm list -a | grep @atproto/
+-- @atproto/ozone@0.1.11 # <- older version !!!
| +-- @atproto/api@0.12.7  #<- even this commit declares @atproto/api as 0.12.22,  binary uses older version
```
